mateog-assign1/crypto.py

```diff
@@ -0,0 +1,322 @@
+        character using map.
+        """
+        encrypt_alphabet = alphabet[3:] + alphabet[:3]
+        print(encrypt_alphabet)
+        encrypt_map = {}
+        for i in range (0, 26):
+                encrypt_map[alphabet[i]] = encrypt_alphabet[i]
+
+        plaintext = plaintext.upper()
+        ciphertext = list(plaintext)
+        for i in range(0, len(plaintext)):
+            ciphertext[i] = encrypt_map[ciphertext[i]]
+
+        return "".join(ciphertext)
+
+
+def decrypt_caesar(ciphertext):
+        """
+        Decrypts a ciphertext using a Caesar cipher. Maps alphabet to plaintext
+        alphabet shifted by three characters to the right, such that A maps to X, etc.
+        Converts ciphertext to list, and finds plaintext version of each
+        character using map.
+        """
+        encrypt_alphabet = alphabet[-3:] + alphabet[:-3]
+        print(encrypt_alphabet)
+        encrypt_map = {}
+        for i in range (0, 26):
+                encrypt_map[alphabet[i]] = encrypt_alphabet[i]
+
+        ciphertext = ciphertext.upper()
+        plaintext = list(ciphertext)
+        for i in range(0, len(ciphertext)):
+            plaintext[i] = encrypt_map[plaintext[i]]
+
+        return "".join(plaintext)
+
+
+def encrypt_vigenere(plaintext, keyword):
+        """
+        Encrypts plaintext using a Vigenere cipher with a keyword. Creates
+        ciphertext list of same length as plaintext, gets ASCII value of
+        each letter in plaintext and corresponding letter in keyword, subtracts
+        value of 'A', adds key character value to plaintext value, mods sum by
+        26 to get a wrapping shift, and adds 'A' back to sum before getting ASCII
+        character back from ciphertext value.
+        """
+        plaintext = plaintext.upper()
+        keyword = keyword.upper()
+        ciphertext = list(plaintext)
+        for i in range(0, len(plaintext)):
+            p = ord(plaintext[i]) - ord('A')
+            k = ord(keyword[i % len(keyword)]) - ord('A')
+            ciphertext[i] = chr(((p + k) % 26) + ord('A'))
+
+        return "".join(ciphertext)
+
+
+def decrypt_vigenere(ciphertext, keyword):
+        """
+        Decrypts ciphertext using a Vigenere cipher with a keyword. Creates
+        plaintext list of same length as ciphertext, gets ASCII value of
+        each letter in ciphertext and corresponding letter in keyword, subtracts
+        value of 'A' from each, subtracts key character value from ciphertext value,
+        mods sum by 26 to get a wrapping shift, and adds 'A' back to sum before
+        getting ASCII character back from plaintext value.
+        """
+        ciphertext = ciphertext.upper()
+        keyword = keyword.upper()
+        plaintext = list(ciphertext)
+        for i in range(0, len(ciphertext)):
+            p = ord(ciphertext[i]) - ord('A')
+            k = ord(keyword[i % len(keyword)]) - ord('A')
+            plaintext[i] = chr(((p - k) % 26) + ord('A'))
```

Comment:
Cool, I just have a nit about using more descriptive variable names - `p` and `k` could be more descriptive about what they store